### DISCUSSION TITLE
Following multi account best practices.

### DISCUSSION THREAD

**Discussion by Xenoha** [awslabs/sbt-aws#45] [General]
Hello! 

As usual, this is some good stuff. It incompasses a lot of best practices for Multi tenant SaaS. And because of your teams experience, the architecture of resources flows nicely.

I'm a little confused on a major best practice that is talked about all the time and used. That is the use of multiple accounts. In practice, services, app planes, control planes, and etc, would all live in other accounts. When are we going to see something that really showcases this major hurdle? 

When using multiple accounts you have networking and resource sharing management, secret sharing management, IAM management for organizations or accounts based on security needs of the services/products, permissions services, and more. One of the major pain points is the SDLC of these different services and deployment of these service. Not to mention if there is a need to build/configure resources for tenants dynamically.

Is the team addressing these challenges through the community with kits like this one?

**Comment by rupetnar** [comment]
@Xenoha Thanks for initiating this discussion. We do have multi-account support on our roadmap. While we don't have a specific timeline to share at the moment, we'll keep the community updated as we make progress on this front. 

We appreciate your interest and feedback. I have moved this discussion to our issues board as a feature enhancement request.

**Comment by suhussai** [comment]
@Xenoha , not sure if you already saw, but we have an RFC that relates to this discussion. It's regarding cell-based architecture, which can in theory, be used to support multi-account SaaS deployments. You can read up on the design and provide feedback here: https://github.com/awslabs/sbt-aws/discussions/146
